maths/fermat_little_theorem.py

```diff
@@ -5,7 +5,7 @@
 # Wikipedia reference: https://en.wikipedia.org/wiki/Fermat%27s_little_theorem
 
 
-def binary_exponentiation(a, n, mod):
+def binary_exponentiation(a: int, n: float | int, mod: int) -> int:
```

Comment:
```suggestion
def binary_exponentiation(a: int, n: float, mod: int) -> int:
```
Per the ruff error, just typehint it as `float` rather than `float | int`

Reply:
@tianyizheng02 thanks for the explanation.